### DISCUSSION TITLE
Is it possible to get the image preview that an article uses?

### DISCUSSION THREAD

**Discussion by luni3359** [barrust/mediawiki#107] [Q&A] [answered]
When you post a Wikipedia link in a site or program that supports "previews", usually a small description will show up, as well as a relevant image (if any). I thought that the first eleent in `page.images` would be the preview, but I was wrong. Aside from doing hacky stuff on `page.html` to retrieve the image preview, it doesn't seem to be possible to get it.

Is there any way to do it that I'm not aware of?

**Comment by barrust** [comment, marked as answer]
After looking a bit at the docs this morning I found [a JSON Endpoint](https://www.mediawiki.org/wiki/Page_Content_Service#JSON_endpoints) that would likely do what you are looking for. There are some portions of this project that attempts to parse the HTML page (`logos`, `hatnotes`, and `parse_section_links`) 

The following code could be used to simulate what I think you are asking:

``` python
import requests

url = 'https://en.wikipedia.org/api/rest_v1/page/summary/Dog'
r = requests.get(url)
print(r.text)
```

If this is what you are looking for, then perhaps this could be added. It isn't exactly the same API and something would need to be done to check if it is available (since this library supports more than just Wikipedia!) but it could be useful. 

Hope this helps!

**Reply by luni3359** [reply]
Thank you, this is exactly what I was looking for! I'll be using this in the meantime. 

As for your suggestion, I'm not sure if you think it'd be worth having this around, but I definitely think this should be. I'd love to contribute and make a PR myself, but I don't think I'm quite there yet in terms of skill.

**Reply by barrust** [reply]
So I have been thinking about this some more; do you think this makes more sense for a page or for the site? In other words, should this be a page property 

```python
# page property
from mediawiki import MediaWiki
wiki = MediaWiki()
page = wiki.page("Washington D.C.")
page.preview  # would be generated when called
```

Or would it be more useful if it was a function of the wiki site? Such as the following:

```python
from mediawiki import MediaWiki
wiki = MediaWiki()
wiki.preview("Washington D.C.")
# probably some sort of tuple response
```

Thoughts?

**Reply by luni3359** [reply]
From my point of view it'd make more sense if the preview came from the page since that's where you can get a summary from as well. Or were you thinking about making ``wiki.preview`` feature a summary of the page, amongst other preview-like properties?

**Reply by barrust** [reply]
Sorry this has been on the back burner; but I pushed a branch that attempts to add this functionality into the system. In the end, I went with a slightly different method in the [PR](https://github.com/barrust/mediawiki/pull/114). I found an example of how it was accomplished prior to the json version and I felt it fit better with what we are doing in pymediawiki. Let me know if this provides what you are looking for!

**Reply by luni3359** [reply]
I don't really know how to test it, but from a quick read it looks good to me.
